### PR TITLE
Only update changed v2.5

### PR DESCRIPTION
**Pull request by rmweir**
**Issue:**
https://github.com/rancher/rancher/issues/37360

### CODE REVIEW

**Comment by rmweir** [comment]
**Testing:**
For QA:
Enabled metrics with controllers
Provision downstream cluster and deploy 10 projects in it with a few mysql apps deployed in each
Can create 10 projects in local cluster (Attempting to creating enough objects where controller spikes are noticeable)
Set CATTLE_RESYNC_DEFAULT rancher env var to 5
confirm that controllers are running every 5 minutes
Set CATTLE_SKIP_NO_CHANGE_UPDATE to mgmt
Confirm handlers spikes every 5 minutes are smaller
Set CATTLE_SKIP_NO_CHANGE_UPDATE to mgmt,user
Should be even smaller
Set CATTLE_SKIP_NO_CHANGE_UPDATE to mgmt,user,scaled
Will probably look comparable or smaller.

For scaling QA:
Provision a scaled setup, something comparable to a benchmark setup, like ~500 clusters
Enable controllers metrics
Set CATTLE_RESYNC_DEFAULT rancher env var to 20
Confirm controller spikes happen on 20 min interval
See if there is an observable memory spike around when controllers spike.
If so, see if Set CATTLE_SKIP_NO_CHANGE_UPDATE set to mgmt appears to reduce it
do the same as above with env var set to mgmt,user

**Comment by rmweir** [comment]
Will update /pkg/apis/go.mod once lasso and norman PRs are approved: https://github.com/rancher/lasso/pull/27 https://github.com/rancher/norman/pull/411

File: pkg/controllers/options.go
Comment (cmurphy) [inline pkg/controllers/options.go:33]: Could this be unexported?

**Review by cmurphy — COMMENTED** [review]
Would an example of using this be to set `CATTLE_SKIP_NO_CHANGE_UPDATE=user,scaled,mgmt` on the Rancher server? Is there a reason to set it on some types of controllers but not all?

File: pkg/controllers/options.go
Comment (rmweir) [reply pkg/controllers/options.go:33]: Yes it can be unexported will change. Gives us a way to isolate which set of controllers are making an impact. Also gives us flexibility if a certain set of controllers ends up having issue.

File: go.sum
Comment (MbolotSuse) [inline go.sum:987]: Similar comment here about the testing libs - leaving this comment to make sure that we get the "testing dependency" stuff removed before we merge this in.

File: pkg/controllers/options.go
Comment (rmweir) [reply pkg/controllers/options.go:24]: I'm not sure if I think this really matter but i went ahead and did it. I fixed this up in the norman PR as well.

File: go.mod
Comment (rmweir) [reply go.mod:12]: Yep, noted here: https://github.com/rancher/rancher/pull/37436#issuecomment-1106985583 and that is why the title has "Do not merge" in it. Will fix this up and the go.sum once other two PR's are ready.

**Review by cmurphy — APPROVED** [review]

**Review by cmurphy — APPROVED** [review]

**Review by MbolotSuse — APPROVED** [review]
LGTM.

**Pull request merged**
